packages/rpcs/wallet_setPreferences/index.js

```diff
@@ -8,28 +8,34 @@ export const schemas = {
     map,
     {closed: true},
     ['hideTestNetwork', {optional: true}, boolean],
-    ['useModernProviderAPI', {optional: true}, boolean],
     ['overrideWindowDotEthereum', {optional: true}, boolean],
-    ['overrideWindowDotConflux', {optional: true}, boolean],
   ],
 }
 
 export const permissions = {
   external: ['popup'],
-  methods: [],
+  methods: ['wallet_getPreferences'],
   db: ['setPreferences', 'getApp'],
 }
 
-export const main = ({db: {setPreferences, getApp}, params}) => {
+export const main = async ({
+  rpcs: {wallet_getPreferences},
+  db: {setPreferences, getApp},
+  params,
+}) => {
   const txRst = setPreferences(params)
 
-  if (!isUndefined(params?.useModernProviderAPI)) {
+  const {overrideWindowDotEthereum} = await wallet_getPreferences()
+
+  // if overrideWindowDotEthereum is set
+  if (!isUndefined(params?.overrideWindowDotEthereum)) {
     const apps = getApp()
     apps.forEach(app => {
```

Comment:
The Dapp pages which does not connected with wallet ,will not receive the post message.
Maybe we should let users render page by themselves?
I have implemented a similar function for your reference (https://github.com/Conflux-Chain/helios/pull/806. Only need to see packages/rpcs/wallet_setPreferences/index.js and  packages/inpage/setup-inpage-provider.js)

Reply:
> he Dapp pages which does not connected with wallet ,will not receive the post message.
> Maybe we should let users render page by themselves?

I don't quite understand why let user manully render all pages when we are able to help them render the connected one.

Reply:
The unconnected dapp does not get message. They still Call old wallet